lib/streamlit/web/bootstrap.py

```diff
@@ -197,6 +197,15 @@ def _fix_pydeck_mapbox_api_warning() -> None:
     os.environ["MAPBOX_API_KEY"] = config.get_option("mapbox.token")
 
 
+def _fix_pydantic_duplicate_validators_error():
+    try:
+        from pydantic import class_validators  # type: ignore[import]
+
+        class_validators.in_ipython = lambda: True
+    except ImportError:
```

Comment:
This function needs a thorough docstring! The context provided in the PR description is great - maybe just copy-paste most of it here?